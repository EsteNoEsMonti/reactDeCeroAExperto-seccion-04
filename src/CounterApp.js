import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Typography } from 'antd';
import { PlusCircleOutlined, MinusCircleOutlined, RetweetOutlined } from '@ant-design/icons';

const { Title } = Typography;

const CounterApp = ({ valor = 1997 }) => {

  // State ---
  const [counter, setCounter] = useState(valor);

  // add reset and remove ---
  const handleAdd = (e) => {
    setCounter(counter + 1);

    //puede recibir una funcion cuyo retorno sea el nuevo valor
    // setCounter((c) => c + 1);
  };

  const handleRemove = (e) => {
    setCounter(counter - 1);
  };

  const handleReset = (e) => {
    setCounter(valor);
  };

  return (
    <>
      <div style={{ textAlign: 'center' }}>
        <Title level={1} style={{ color: 'white' }}>Counter App</Title>
        <Title level={2} style={{ color: 'white' }} id="test_counter">{counter}</Title>

        <Button
          type="default"
          onClick={handleRemove}
          style={{ margin: 15 }}
        >
          <MinusCircleOutlined />1
              </Button>
        <Button
          ghost
          onClick={handleReset}
          style={{ margin: 15 }}
        >
          <RetweetOutlined /> Reset
              </Button>
        <Button
          type="primary"
          onClick={handleAdd}
          style={{ margin: 15 }}
        >
          <PlusCircleOutlined />1
              </Button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  valor: PropTypes.number
};

export default CounterApp;







// import React, { useState/*, useEffect*/ } from 'react';
// import PropTypes from 'prop-types';
// import { Button, Typography/*, message, Card, Form, InputNumber*/ } from 'antd';
// import { PlusCircleOutlined, MinusCircleOutlined, RetweetOutlined } from '@ant-design/icons';

// // const layout = {
// //   labelCol: {
// //     span: 8,
// //   },
// //   wrapperCol: {
// //     span: 16,
// //   },
// // };
// // const tailLayout = {
// //   wrapperCol: {
// //     offset: 8,
// //     span: 16,
// //   },
// // };

// const { Title } = Typography;

// const CounterApp = ({ valor = 1997 }) => {

//   // State ---
//   // const [showCounter, setShowCounter] = useState(false);
//   const [counter, setCounter] = useState(valor);
//   // const [valorForm, setValorForm] = useState(valor);

//   // const [form] = Form.useForm();

//   // add reset and remove ---
//   const handleAdd = (e) => {
//     setCounter(counter + 1);

//     //puede recibir una funcion cuyo retorno sea el nuevo valor
//     // setCounter((c) => c + 1);
//   };

//   const handleRemove = (e) => {
//     setCounter(counter - 1);
//   };

//   const handleReset = (e) => {
//     // setCounter(valorForm);
//     setCounter(valor);
//   };

//   // // Fn form ---
//   // const onFinish = (values) => {
//   //   setCounter(values.valueInput);
//   //   setValorForm(values.valueInput);

//   //   form.setFieldsValue({
//   //     valueInput: values.valueInput
//   //   });

//   //   setShowCounter(true);
//   //   message.success('Valor seteado correctamente, disfrute su contador', 10);
//   // };

//   // useEffect(() => {
//   //   if (valor) {
//   //     form.setFieldsValue({
//   //       valueInput: valor
//   //     });
//   //   };
//   // });

//   return (
//     // <>
//     //   {
//     //     !showCounter ?
//     //       <>
//     //         <Title level={4} style={{ color: 'white' }}>El valor por defecto se lo pasa como prop desde el componente padre</Title>
//     //         <Card>
//     //           <Form
//     //             {...layout}
//     //             name="basic"
//     //             initialValues={{
//     //               remember: true,
//     //             }}
//     //             onFinish={onFinish}
//     //             form={form}
//     //           >
//     //             <Form.Item
//     //               label="Ingrese un valor"
//     //               name="valueInput"
//     //               rules={[
//     //                 {
//     //                   required: true,
//     //                   message: 'Please input your value!',
//     //                 },
//     //               ]}
//     //             >
//     //               <InputNumber style={{ width: '100%' }} />
//     //             </Form.Item>

//     //             <Form.Item {...tailLayout}>
//     //               <Button type="primary" htmlType="submit">
//     //                 Enviar valor
//     //             </Button>
//     //             </Form.Item>

//     //           </Form>
//     //         </Card>
//     //       </>
//     //       :
//     <>
//       <div style={{ textAlign: 'center' }}>
//         <Title level={1} style={{ color: 'white' }}>Counter App</Title>
//         <Title level={2} style={{ color: 'white' }}>{counter}</Title>

//         <Button
//           type="default"
//           onClick={handleRemove}
//           style={{ margin: 15 }}
//         >
//           <MinusCircleOutlined />1
//               </Button>
//         <Button
//           ghost
//           onClick={handleReset}
//           style={{ margin: 15 }}
//         >
//           <RetweetOutlined /> Reset
//               </Button>
//         <Button
//           type="primary"
//           onClick={handleAdd}
//           style={{ margin: 15 }}
//         >
//           <PlusCircleOutlined />1
//               </Button>
//       </div>
//     </>
//     //   }
//     // </>
//   );
// };

// CounterApp.propTypes = {
//   valor: PropTypes.number
// };

// export default CounterApp;