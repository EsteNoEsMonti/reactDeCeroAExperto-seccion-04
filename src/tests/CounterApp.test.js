import React from 'react';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp/>', () => {
  
  //let wrapper; // undefined, pierdo el intellisense
  let wrapper = shallow(<CounterApp />); //no es buena practica pero no pierno el intellisense
  beforeEach( () => {
    wrapper = shallow(<CounterApp />); //para que se ejecute en cada test
  });

  test('Debe mostrar CounterApp correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar el valor por efecto de 420', () => {

    const wrapper = shallow(
      <CounterApp
        valor={420}
      />
    );

    const valorHTML = wrapper.find('#test_counter').html();
    expect(valorHTML.includes('420')).toBe(true);
  });

  test('Debe funcionar el btn +1', () => {
    wrapper.find('Button').at(2).simulate('click'); //va el evento que uqiero simular y los valores del evento
    const valorHTML = wrapper.find('#test_counter').html();
    expect(valorHTML.includes('1998')).toBe(true);
  });

  test('Debe funcionar el btn -1', () => {
    wrapper.find('Button').at(0).simulate('click');
    const valorHTML = wrapper.find('#test_counter').html();
    expect(valorHTML.includes('1996')).toBe(true);
  });

  test('Debe funcionar el btn Reset', () => {

    const wrapper = shallow(
      <CounterApp
        valor={420}
      />
    );
    
    wrapper.find('Button').at(2).simulate('click'); //debo modificar el valor par asegurarme que se resetee
    wrapper.find('Button').at(1).simulate('click');
    
    const valorHTML = wrapper.find('#test_counter').html();
    expect(valorHTML.includes('420')).toBe(true);
  });

});