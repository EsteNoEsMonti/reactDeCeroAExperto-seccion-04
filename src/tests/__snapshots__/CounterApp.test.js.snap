// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> Debe mostrar CounterApp correctamente 1`] = `
<Fragment>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <Title
      level={1}
      style={
        Object {
          "color": "white",
        }
      }
    >
      Counter App
    </Title>
    <Title
      id="test_counter"
      level={2}
      style={
        Object {
          "color": "white",
        }
      }
    >
      1997
    </Title>
    <Button
      onClick={[Function]}
      style={
        Object {
          "margin": 15,
        }
      }
      type="default"
    >
      <ForwardRef(MinusCircleOutlined) />
      1
    </Button>
    <Button
      ghost={true}
      onClick={[Function]}
      style={
        Object {
          "margin": 15,
        }
      }
    >
      <ForwardRef(RetweetOutlined) />
       Reset
    </Button>
    <Button
      onClick={[Function]}
      style={
        Object {
          "margin": 15,
        }
      }
      type="primary"
    >
      <ForwardRef(PlusCircleOutlined) />
      1
    </Button>
  </div>
</Fragment>
`;
